### DISCUSSION TITLE
Basic saving anvil chunks to world folder

### DISCUSSION THREAD

**Discussion by JackM33** [Minestom/Minestom#623] [Help] [answered]
(Posting this though I believe I'm on the right track but for future reference for people)

Hi I'm obviously totally new to Minestom and trying to save chunks to the disk with a call to saveChunksToStorage. I've started with the basic server set up from the [wiki](https://wiki.minestom.net/setup/your-first-server) . As I understand it I obviously need to load the chunks before saving them. Is that event handled as I join the world or is there a default loading that happens as I understood a traditional MC server would near the spawn region? Otherwise how should I initiate the process of saving chunks.

[Updated] I think I understand I need to have a scheduled call to saveChunksToStorage happen asynchronously to the server? Thanks!

**Comment by JackM33** [comment, marked as answer]
I've addressed this issue by saving chunks to storage with a schedule manager. (saves every 20 seconds)

```
public static void main(String[] args) {
        // Initialization
        MinecraftServer minecraftServer = MinecraftServer.init();

        InstanceManager instanceManager = MinecraftServer.getInstanceManager();
        // Create the instance
        InstanceContainer instanceContainer = instanceManager.createInstanceContainer();

        MinecraftServer.getSchedulerManager()
                .buildTask(() -> {
                    instanceContainer.saveChunksToStorage();
                })
                .delay(20, ChronoUnit.SECONDS)
                .schedule();

        // Set the ChunkGenerator
        instanceContainer.setChunkGenerator(new GeneratorDemo());

        // Add an event callback to specify the spawning instance (and the spawn position)
        GlobalEventHandler globalEventHandler = MinecraftServer.getGlobalEventHandler();
        globalEventHandler.addListener(PlayerLoginEvent.class, event -> {
            final Player player = event.getPlayer();
            event.setSpawningInstance(instanceContainer);
            player.setRespawnPoint(new Pos(0, 42, 0));
        });


        // Start the server on port 25565
        minecraftServer.start("0.0.0.0", 25565);

    }
```

**Reply by TheMode** [reply]
Be aware that the `delay` schedule is only used once (the first time). You are probably looking for `repeat` (maybe both, start after 20 seconds, and then every 20 seconds) https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/timer/Task.java#L63

Must also mention that tasks are executed sync by default, you can make them async by changing their `ExecutionType`. Though by that point, if you need an async task scheduled based on time, you will be better off using a [ScheduledExecutorService](https://www.baeldung.com/java-executor-service-tutorial#ScheduledExecutorService)
